src/test/java/net/datafaker/script/ProvidersDocsGenerator.java

```diff
@@ -0,0 +1,191 @@
+        providersDocsGenerator.generateProvidersDocs();
+    }
+
+    public void generateProvidersDocs() {
+        fillTreeSet();
+
+        Set<String> fakersWithoutSinceTag = new HashSet<>();
+        for (Class<?> clazz : subTypes) {
+            String name = clazz.getSimpleName();
+
+            if (providersToExcludeFromGeneration.contains(name)) {
+                continue;
+            }
+
+            String tag;
+            try {
+                tag = extractTagFromJavadoc("src/main/java/net/datafaker/" + name + ".java", fakersWithoutSinceTag);
+            } catch (FileNotFoundException e) {
+                throw new RuntimeException("Error in supplied path to provider", e);
+            }
+
+            String sinceTag = matchSinceTag(tag);
+            writeProviderToTable(name, sinceTag);
+        }
+        System.out.println("Providers without '@since' tag: " + fakersWithoutSinceTag);
+    }
+
+    /**
+     * The implementation assumes that the JavaDoc with the '@since' tag
+     * will always be at the very beginning of a class.
+     *
+     * @param filePath                              Path to the Faker(provider) to be searched
+     * @param fakersWithoutSinceTag                 List in which add Faker's names that don't have '@since' tag
+     * @return Entire first JavaDoc in the class
+     * @throws FileNotFoundException
+     */
+    private String extractTagFromJavadoc(String filePath, Set<String> fakersWithoutSinceTag) throws FileNotFoundException {
+        final File file = new File(filePath);
+
+        Optional<CommentsCollection> commentsCollection = parser.parse(file).getCommentsCollection();
+
+        if (!commentsCollection.isPresent()) {
+            fakersWithoutSinceTag.add(filePath);
+            return "";
+        }
+
+        Optional<JavadocComment> javadocComments = commentsCollection.get().getJavadocComments()
+            .stream()
+            .findFirst();
+
+        if (!javadocComments.isPresent()) {
+            fakersWithoutSinceTag.add(filePath);
+            return "";
+        }
+
+        String comment = javadocComments.get().getContent();
+        boolean isComment = comment.contains("@since");
+
+        if (!isComment) {
+            fakersWithoutSinceTag.add(filePath);
+        }
+
+        return comment;
+    }
+
+    /**
+     * Writes header and table header to new 'providers.md'
+     *
+     * @see #writeToFile(String)
+     */
+    private void constructHeaderInProvidersFile() {
+        final String header = "# Fake Data Providers\n" + "\n" + "Datafaker comes with the following list of data providers:" + "\n";
+        final String tableHeader = "\n" + "| Name" + Strings.repeat(" ", 88) + "|" + " " + "Description" + Strings.repeat(" ", 120) + "|" + " " + "Since" + " " + "|" + "\n";
+        final String tableHeaderDelimiter = "|" + Strings.repeat("-", 93) + "|" + Strings.repeat("-", 132) + "|" + Strings.repeat("-", 7) + "|";
```

Comment:
```suggestion
    public String generateColumn(String name, char padSymbol, int length) {
        if (name.length() >= length) return name;
        return padSymbol + name + Strings.repeat(padSymbol, length - name.length() - 1);
    }
    /**
     * Writes header and table header to new 'providers.md'
     *
     * @see #writeToFile(String)
     */
    private void constructHeaderInProvidersFile() {
        final int nameColLength = 93;
        final int descColLength = 132;
        final int sinceColLength = 7;
        final String header = "# Fake Data Providers\n" + "\n" + "Datafaker comes with the following list of data providers:" + "\n";
        final String tableHeader = "\n|" + generateColumn("Name", ' ', nameColLength) + "|" + generateColumn("Description", ' ', descColLength) + "|" + generateColumn("Since", ' ', sinceColLength) + "|\n";
        final String tableHeaderDelimiter = "|" + generateColumn("", '-', nameColLength) + "|" + generateColumn("", '-', descColLength)  + "|" + generateColumn("", '-', sinceColLength) + "|";
```